### PR TITLE
core(core-flows, medusa): remove request item return

### PR DESCRIPTION
**Pull request by carlos-r-l-rodrigues**
What:
* `DELETE /admin/returns/:id/request-items/:action_id`
* add missing `deleted_at` columns

FIXES: CC-188

### CODE REVIEW

File: packages/modules/order/src/migrations/Migration20240715174100.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { Migration } from "@mikro-orm/migrations"
 
-export class Migration20240715102100 extends Migration {
+export class Migration20240715174100 extends Migration {
```
Comment (olivermrbl) [inline packages/modules/order/src/migrations/Migration20240715174100.ts:3]: question: should we revert this change?

**Review by olivermrbl — APPROVED** [review]
LGTM!

File: packages/modules/order/src/migrations/Migration20240715174100.ts (hunk 1 above)
Comment (carlos-r-l-rodrigues) [reply packages/modules/order/src/migrations/Migration20240715174100.ts:3]: I think the migration will read the file name instead of the Class. In the previous commit I didn't change the name and it was the same as the previous migration at that time.
Should be fine to keep it
